File: 1.py
#Project Euler solution 1, author: http://github.com/TerkaSlan #
__author__ = 'terka'

from time import clock

start = clock()
sum=0
for i in range(3,1000,3):
    sum += i
for i in range(5,1000,5):
    sum += i
for i in range(15,1000,15):
    sum -= i
end = clock()

print 'Found a solution: ', sum
print 'It took me ', end-start, ' seconds' #0.000106  seconds

File: 1.c
/*Project Euler solution 1, author: http://github.com/TerkaSlan */
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define LIMIT 1000
int main()
{
    clock_t start, end;
    float time_passed;

    short unsigned i;
    unsigned sum=0;
    start = clock();
    for(i=3; i<LIMIT; i+=3){
        sum += i;
    }
    for(i=5; i<LIMIT; i+=5){
        sum += i;
    }
    //Subtracting multiples of 15, since they were counted twice
    for(i=15; i<LIMIT; i+=15){
        sum -=i;
    }
    end = clock();

    printf("\nI found a solution: %u\n", sum);

    time_passed = (end-start)*1.0/CLOCKS_PER_SEC;
    printf("\nThe execution took me %f seconds\n", time_passed); // 0.000007 seconds
    return 0;
}

